src/pybind/mgr/prometheus/module.py

```diff
@@ -1294,15 +1296,17 @@ def get_rbd_stats(self) -> None:
                 namespace_name = s[1]
 
             if pool_name == "*":
-                # collect for all pools
-                osd_map = self.get('osd_map')
-                for pool in osd_map['pools']:
+                # collect for all existing pools
+                for pool in existing_pools:
                     if 'rbd' not in pool.get('application_metadata', {}):
                         continue
                     pool_keys.append((pool['pool_name'], namespace_name))
             else:
-                pool_keys.append((pool_name, namespace_name))
+                for pool in existing_pools:
+                    if pool['pool_name'] == pool_name:
+                        pool_keys.append((pool_name, namespace_name))  # avoids adding deleted pool
 
+        pool_keys = list(set(pool_keys))  # removing duplicate entries
```

Comment:
I'd apply this reduction before the for-loop: if a user specifies `rbd_stats_pools = rbd,rbd,rbd`, we should deduplicate asap.